tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -229,30 +229,51 @@ void test_variables_construction_and_access() noexcept {
       == "Variables(vector,scalar,scalar2)");
 }
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables.Move",
-                  "[DataStructures][Unit]") {
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> x(1, -2.0),
-      z(2, -3.0);
-  CHECK(&get<VariablesTestTags_detail::vector>(z)[0][0] == z.data());
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> y = std::move(x);
-  x = std::move(z);
-  CHECK(
-      (x == Variables<tmpl::list<VariablesTestTags_detail::vector>>{2, -3.0}));
-  CHECK(&get<VariablesTestTags_detail::vector>(x)[0][0] == x.data());
+template <typename VectorType>
+void test_variables_move() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<
+      typename tt::get_fundamental_type_t<typename VectorType::value_type>>
+      dist{-100, 100};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t number_of_grid_points1 = sdist(gen),
```

Comment:
`const`? We prefer specifying the type on both lines (something I disagreed with for a long time but have come to appreciate a lot since reading other people's code more)